kurl_proxy/cmd/main.go

```diff
@@ -270,6 +270,34 @@ func getHttpServer(fingerprint string, acceptAnonymousUploads bool, assetsDir st
 
 		app, err := kotsadmApplication()
 
+		if err != nil {
+			log.Printf("No kotsadm application metadata: %v", err) // continue
+		}
+
+		appIcon := template.URL(app.Spec.Icon)
+		c.HTML(http.StatusOK, "welcome.html", gin.H{
+			"fingerprintSHA1":   fingerprint,
+			"AppIcon":           appIcon,
+			"AppTitle":          app.Spec.Title,
+			"IsEmbeddedCluster": isEmbeddedCluster(),
+		})
+	})
+	r.GET("/insecure", func(c *gin.Context) {
```

Comment:
since everything else is the same, you can do something like this instead:
```golang
	r.GET("/", func(c *gin.Context) {
...
		appIcon := template.URL(app.Spec.Icon)
		htmlPage := "welcome.html"
		if c.Request.URL.Path == "/insecure" {
			htmlPage = "insecure.html"
		}
		c.HTML(http.StatusOK, htmlPage, gin.H{
			"fingerprintSHA1":   fingerprint,
			"AppIcon":           appIcon,
			"AppTitle":          app.Spec.Title,
			"IsEmbeddedCluster": isEmbeddedCluster(),
		})
...
```